test/integration/scheduler_perf/util.go

```diff
@@ -381,7 +391,8 @@ func newThroughputCollector(podInformer coreinformers.PodInformer, labels map[st
 	}
 }
 
-func (tc *throughputCollector) init() {
+func (tc *throughputCollector) init() error {
+	return nil
```

Comment:
A bit unrelated to this PR though, don't we actually have to stop the throughput collector here?

Looks like it runs until a given ctx is done:
https://github.com/kubernetes/kubernetes/blob/0b8e5f164e452132e57a31b65538ec1fa28b2ffd/test/integration/scheduler_perf/util.go#L412-L413

And, a given ctx is actually done when the test finishes:
https://github.com/kubernetes/kubernetes/blob/0b8e5f164e452132e57a31b65538ec1fa28b2ffd/test/integration/scheduler_perf/scheduler_perf.go#L1159-L1160

meaning, we don't stop the throughput collector until the test finishes once it gets started.

Reply:
I don't see what's an issue here. Could you rephrase?

Reply:
Sorry, nvm. We don't have any cases causing issues with the current impl given we basically don't allow (and actually performance-config.yaml doesn't have any test cases with) multiple `collectMetrics: true`.